pipeline/splicing/calcVarPriors.py

```diff
@@ -139,22 +168,314 @@ def getVarType(variant):
+    for exon in varExons.keys():
+        exonEnd = int(varExons[exon]["exonEnd"])
+        if varStrand == "+":
+            donorStart = exonEnd - 3 + 1
+            donorEnd = exonEnd + 6
+        else:
+            donorStart = exonEnd + 3
+            donorEnd = exonEnd - 6 + 1
+        donorBoundaries[exon] = {"donorStart": donorStart,
+                                 "donorEnd": donorEnd}
+
+    return donorBoundaries
 
 
-def getVarDict(variant):
+def getRefSpliceAcceptorBoundaries(variant):
+    '''
+    Given a variant, returns the splice acceptor boundaries
+    (splice acceptor region is 20 bases before exon and first 3 bases in exon)
+    for the variant's transcript in a dictionary with the format:
+    key = exon number, value = a dictionary with acceptor start and acceptor end for exon
+    '''
+    varExons = getExonBoundaries(variant)
+    if variant["Gene_Symbol"] == "BRCA1" or variant["Gene_Symbol"] == "BRCA2":
+        del varExons["exon1"]
+    varStrand = getVarStrand(variant)
+    acceptorBoundaries = {}
+    for exon in varExons.keys():
+        exonStart = int(varExons[exon]["exonStart"])
+        if varStrand == "+":
+            acceptorStart = exonStart - 20 + 1
+            acceptorEnd = exonStart + 3
+        else:            
+            acceptorStart = exonStart + 20
+            acceptorEnd = exonStart - 3 + 1
+        acceptorBoundaries[exon] = {"acceptorStart": acceptorStart,
+                                    "acceptorEnd": acceptorEnd}
+
+    return acceptorBoundaries
+
+
+def varInExon(variant):
+    '''
+    Given a variant, determines if variant genomic position is inside transcript boundaries
+    AND if variant is in an exon
+    Returns true if variant is in an exon
+    '''
+    varGenPos = int(variant["Pos"])
+    varExons = getExonBoundaries(variant)
+    varStrand = getVarStrand(variant)
+    varOutBounds = varOutsideBoundaries(variant)
+    if varOutBounds == False:
+        for exon in varExons.keys():
+            exonStart = varExons[exon]["exonStart"]
+            exonEnd = varExons[exon]["exonEnd"]
+            if varStrand == "+":
+                if varGenPos > exonStart and varGenPos <= exonEnd:
+                    return True
+            else:
+                if varGenPos <= exonStart and varGenPos >= exonEnd:
+                    return True
+        return False
+    else:
+        return "varOutBounds"    
+
+    
+def varInSpliceDonor(variant):
+    '''
+    Given a variant, determines if a variant is in reference transcript's splice donor region
+    splice donor region = last 3 bases in exon and first 6 bases in intron
+    Returns True if variant in splice donor region
+    '''
+    varGenPos = int(variant["Pos"])
+    varStrand = getVarStrand(variant)
+    donorBounds = getRefSpliceDonorBoundaries(variant)
+    for exon in donorBounds.keys():
+        donorStart = donorBounds[exon]["donorStart"]
+        donorEnd = donorBounds[exon]["donorEnd"]
+        if varStrand == "+":
```

Comment:
I could add a function, similar to the one below:
```
def checkWithinBoundaries(varStrand, varGenPos, boundaryStart, boundaryEnd):
    if varStrand == "+":
        if varGenPos >= boundaryStart and varGenPos <= boundaryEnd:
            return True
```
etc.
But I think I could really only use this on 2 functions (varInSpliceDonor and varInSpliceAcceptor). I could potentially use it in varInExon but would then have to add another argument to the function about whether to use  >=  or  > 
I'm happy to add that in if you think it would be helpful

Reply:
up to you! either way's fine

Reply:
I added the function, definitely makes it easier to read and makes so I don't repeat myself